mos-platform/fds/mapper.c

```diff
@@ -0,0 +1,62 @@
+#include "mapper.h"
+#include <peekpoke.h>
+#include <stdbool.h>
+
+#define FDS_IO 0x4023
+
+#define FDS_WRITE 0x4024
+
+#define FDS_CTRL 0x4025
+#define FDS_CTRL_MIRROR 0xFA
+
+#define FDS_EXT 0x4026
+#define FDS_EXT_MIRROR 0xF9
+
+#define IO_DISK 0x01
+#define IO_SOUND 0x02
+
+#define CTRL_RESET_TRANSFER 0x01
+#define CTRL_TRANSFER_MODE 0x04
+#define CTRL_MIRRORING 0x08
+
+__attribute__((section(".zp.bss"))) volatile char _IO_ENABLED;
+
+__attribute__((leaf)) void fds_set_disk_io(bool value) {
```

Comment:
Missed this the first time around: you don't need leaf on functions that are written in C. You can remove it from the header and the implementation. This is only useful in the headers for functions defined in assembly; otherwise, the compiler can easily tell it's a leaf.